uint8 mode_val
float64[3] goal_pos